### PR TITLE
make `{Index, IndexMut}` take their keys by value

### PR DESCRIPTION
**Pull request by apasel422**

### CODE REVIEW

**Comment by FlashCat** [comment]
r? @reem

(I've picked a reviewer for you, use r? to override)

File: src/lib.rs [hunk 1]
```diff
@@ -161,15 +161,15 @@ impl<K: Hash + Eq, V, S: HashState> LinkedHashMap<K, V, S> {
     /// # Examples
     ///
     /// ```
-    /// # extern crate "linked-hash-map" as linked_hash_map;
+    /// # extern crate linked_hash_map;
```
Comment (Gankro) [inline src/lib.rs]: Are the externs necessary anymore?

Comment (apasel422) [inline src/lib.rs]: Alas, yes. I'm filing a bug now.

File: src/lib.rs [hunk 2]
```diff
@@ -913,8 +913,8 @@ mod tests {
             assert_eq!(None, iter.next());
         }
 
-        assert_eq!(17, map["a"]);
-        assert_eq!(23, map["b"]);
+        assert_eq!(17, map[&"a"]);
+        assert_eq!(23, map[&"b"]);
```
Comment (Gankro) [inline src/lib.rs]: aw, that sucks. &str maps are probably exceptional anyway.

Comment (apasel422) [inline src/lib.rs]: This is actually the result of another bug: https://github.com/rust-lang/rust/issues/23649. We should be able to go back to `map["a"]` once it's fixed.

Comment (Gankro) [inline src/lib.rs]: Really? It makes sense to me. It's a Map<&str, i32>. Seems to follow the &&str is necessary?

Comment (apasel422) [inline src/lib.rs]: `Index` and `get` use `Borrow`, and `&str: Borrow<str>`. (For what it's worth, the crate type-checks with just `map["a"]`, but that bug prevents it from being built.)

Comment (Gankro) [inline src/lib.rs]: Hmm, I might be thinking back to BorrowFrom. Cool that that works!

**Pull request merged**